File: 1/three-and-five.py
#!/usr/bin/env python3
#-*- coding: utf-8 -*-

import sys

def main():
    if len(sys.argv) != 2:
        return help_me(
            "ERR This takes a single " + 
            "integer as an argument.")
    # need to cast sys.argv[1] as it comes in as a string
    hits = [i for i in range(1, int(sys.argv[1])) 
            if is_multiple(3, i) or 
            is_multiple(5, i)]
    res = 0
    for i in hits:
        res += i
    print(res)
    return 0

def help_me(err=None):
    print(err)

    print("Usage:")
    print(f"\t{sys.argv[0]} integer")
    print(
        "Display sum of multiples of 3 and 5",
        "between 1 to the specified integer."
    )
    return 4

def is_multiple(this, that):
    '''determine if this is a multiple fo that.'''
    status = False
    if that % this == 0:
        status = True
    return status

if __name__ == '__main__':
    sys.exit(main())

File: 1/three-and-five.go
package main

import (
	"fmt"
	"os"
	"strconv" 	// the command line args give us a string
			  	// but we need an integer
)

func isMultiple(this int, that int) bool {
	state := false
	if that%this == 0 {
		state = true
	}
	return state
}

func helpMe(errmsg string) int {
	fmt.Println(errmsg)
	fmt.Println("Usage:")
	fmt.Printf("%s integer\n", os.Args[0])
	fmt.Printf("Display sum of multiples of 3 and 5")
	fmt.Printf(" between 1 to the specified integer\n")
	return 5
}

func main() {
	if len(os.Args) != 2 {
		os.Exit(helpMe("ERR This takes a single integer as an argument"))
	}
	counter, _ := strconv.Atoi(os.Args[1])
	res := 0
	for i :=1 ; i < counter ; i ++ {
		if ( isMultiple(3, i) || isMultiple(5, i) ) {
			res += i
		}
	}
	fmt.Println(res)
}